source/priority_queue.c

```diff
@@ -206,6 +210,14 @@ bool aws_priority_queue_is_valid(const struct aws_priority_queue *const queue) {
     return pred_is_valid && container_is_valid && backpointer_list_is_valid && backpointer_struct_is_valid;
 }
 
+bool aws_backpointer_index_valid(const struct aws_priority_queue *const queue, size_t index) {
+    if (queue->backpointers.data && index < queue->backpointers.length) {
+        struct aws_priority_queue_node *node = ((struct aws_priority_queue_node **)queue->backpointers.data)[index];
```

Comment:
Is the cast to `aws_priority_queue_node **` (rather than just single `*`) correct?

Reply:
Thanks for the comment. From my understanding the backpointers array list holds pointers to `struct aws_priority_queue_node`. So the backpointers.data should be a double pointer. Does this sound reasonable?